client/src/pages/user/Profile.test.js

```diff
@@ -0,0 +1,329 @@
+    return {
+        matches: false,
+        addListener: function() {},
+        removeListener: function() {}
+    };
+};
+
+describe('Profile Page', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        window.localStorage.clear();
+        window.localStorage.setItem(
+            "auth",
+            JSON.stringify({ user: baseAuth.user, token: baseAuth.token })
+        );
+        useAuth.mockReturnValue([baseAuth, mockSetAuth])
+    });
+
+    it("renders inputs prefilled from auth.user and email input is disabled", async () => {
+        const { findByPlaceholderText } = render(
+            <MemoryRouter initialEntries={['/dashboard/user/profile']}>
+                <Routes>
+                    <Route path="/dashboard/user/profile" element={<Profile />} />
+                </Routes>
+            </MemoryRouter>
+        )
+
+        const name = await findByPlaceholderText(/enter your name/i);
+        const email = await findByPlaceholderText(/enter your email/i);
+        const phone = await findByPlaceholderText(/enter your phone/i);
+        const address = await findByPlaceholderText(/enter your address/i);
+
+        expect(name).toHaveValue(baseAuth.user.name);
+        expect(email).toHaveValue(baseAuth.user.email);
+        expect(email).toBeDisabled();
+        expect(phone).toHaveValue(baseAuth.user.phone);
+        expect(address).toHaveValue(baseAuth.user.address);
+    });
+
+    it("updates local input state when the user types", async () => {
+        const { findByPlaceholderText } = render(
+            <MemoryRouter initialEntries={['/dashboard/user/profile']}>
+                <Routes>
+                    <Route path="/dashboard/user/profile" element={<Profile />} />
+                </Routes>
+            </MemoryRouter>
+        )
+
+        const name = await findByPlaceholderText(/enter your name/i);
+        const phone = await findByPlaceholderText(/enter your phone/i);
+        const address = await findByPlaceholderText(/enter your address/i);
+        const password = await findByPlaceholderText(/enter your password/i);
+
+        await act( async () => {
+            await userEvent.clear(name);
+            await userEvent.type(name, newInputValues.name);
+            await userEvent.clear(phone);
+            await userEvent.type(phone, newInputValues.phone);
+            await userEvent.clear(address);
+            await userEvent.type(address, newInputValues.address);
+            await userEvent.clear(password);
+            await userEvent.type(password, newInputValues.password);
+        })
+
+        expect(name).toHaveValue(newInputValues.name);
+        expect(phone).toHaveValue(newInputValues.phone);
+        expect(address).toHaveValue(newInputValues.address);
+        expect(password).toHaveValue(newInputValues.password);
+    });
+
+    it('Successful Submission updates Auth, Local Storage and Shows Success Toast', async () => {
```

Comment:
Using `it()`, the test case name should follow a format where the entire line reads like a natural sentence, e.g. `it("renders ...)` or `it("should ...")`
Here, a better format would be
```suggestion
    it("updates auth, local storage and shows success toast upon successful submission", async () => {
```